cypress/integration/main-page_spec.js

```diff
@@ -0,0 +1,63 @@
+describe('main page', () => {
+  beforeEach(() => {
+    cy.fixture('./stocks.json').then((allStocks) => {
+      cy.intercept('GET', 'https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=30&page=1&sparkline=false', {
+        statusCode: 200,
+        body: allStocks
+      })
+    cy.visit('http://localhost:3000/');
+    })
+  })
+
+  it('should contain a header', () => {
+    cy.contains('h1', 'MARKET WATCH')
+  })
+
+  it('should render stock cards upon page visit', () => {
+    cy.get('.stock')
```

Comment:
Have you looked into the data-cy labels in the docs? It has best practices for labeling using cypress. For instance, instead of using the element or className, the elements have their own attribute of data-cy="label-goes-here" and then you import it as cy.get('[data-cy=label-goes-here]'). I haven't heard feedback regarding this really, but it makes sense per the document suggestions. Just a thought.